TypeError: Cannot use 'in' operator to search for 'start' in undefined

### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

"eslint-plugin-mdx": "^3.1.4"

### Link to runnable example

https://codesandbox.io/p/devbox/eslint-plugin-mdx-typeerror-remark-lint-xqwl2v

### Steps to reproduce

Steps to reproduce:

1. Start with [MDX Loader with Next.js](https://codesandbox.io/s/github/mdx-js/.github/tree/main/sandbox-templates/mdx-loader-next) codesandbox from issue template.
2. Install `eslint` and `eslint-mdx`
```bash
npm i -D eslint-plugin-mdx eslint
```
3. Add eslint config file `.eslintrc`. Config pulled [directly from readme](https://github.com/mdx-js/eslint-mdx?tab=readme-ov-file#classic).
```
{
  "extends": ["plugin:mdx/recommended"],
  "settings": {
    "mdx/code-blocks": true,
    "mdx/language-mapper": {}
  }
}
```
4. Install `remark-lint`. [reference](https://github.com/remarkjs/remark-lint?tab=readme-ov-file#example-check-markdown-on-the-api)
```bash
npm install vfile-reporter remark remark-preset-lint-consistent remark-preset-lint-recommended remark-lint-list-item-indent --save-dev
```
5. Setup remark config by adding `remarkConfig` to `package.json`. Config pulled directly [from remark-lint readme](https://github.com/remarkjs/remark-lint?tab=readme-ov-file#example-check-markdown-on-the-cli).
6. Remove the trailing new line from `pages\index.mdx`.
7. Run `npx eslint . --ext mdx` and result is:
```bash
/workspace git:(master) ✗ npx eslint . --ext mdx

Oops! Something went wrong! :(

ESLint: 8.56.0

TypeError: Cannot use 'in' operator to search for 'start' in undefined
Occurred while linting /workspace/pages/index.mdx:1
Rule: "mdx/remark"
    at Program (/workspace/node_modules/eslint-plugin-mdx/lib/rules/remark.js:67:38)
    at ruleErrorHandler (/workspace/node_modules/eslint/lib/linter/linter.js:1076:28)
    at /workspace/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/workspace/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/workspace/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/workspace/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/workspace/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/workspace/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:803:23)
    at /workspace/node_modules/eslint/lib/linter/linter.js:1111:32
```



### Expected behavior

eslint should output:

```bash
/workspace git:(master) ✗ npx eslint . --ext mdx

/workspace/pages/index.mdx
  1:1  warning  Missing newline character at end of file  remark-lint-final-newline

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.
```

### Actual behavior

eslint outputs:

```bash
/workspace git:(master) ✗ npx eslint . --ext mdx

Oops! Something went wrong! :(

ESLint: 8.56.0

TypeError: Cannot use 'in' operator to search for 'start' in undefined
Occurred while linting /workspace/pages/index.mdx:1
Rule: "mdx/remark"
    at Program (/workspace/node_modules/eslint-plugin-mdx/lib/rules/remark.js:67:38)
    at ruleErrorHandler (/workspace/node_modules/eslint/lib/linter/linter.js:1076:28)
    at /workspace/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/workspace/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/workspace/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/workspace/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/workspace/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/workspace/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:803:23)
    at /workspace/node_modules/eslint/lib/linter/linter.js:1111:32
```

### Runtime

Node v20

### Package manager

npm v9

### OS

Linux

### Build and bundle tools

Next.js

--- a/.changeset/new-brooms-visit.md
+++ b/.changeset/new-brooms-visit.md
@@ -1,0 +1,10 @@
+---
+"eslint-plugin-mdx": patch
+---
+
+fix: `undefined` `place` in vFile message causes TypeError
+
+Many `remark-lint` errors do not set a place variable in the vFile
+message. This code should accept `undefined` for this field.
+
+fixes #520

--- a/packages/eslint-plugin-mdx/src/rules/remark.ts
+++ b/packages/eslint-plugin-mdx/src/rules/remark.ts
@@ -100,7 +100,7 @@ export const remark: Rule.RuleModule = {
             // related to https://github.com/eslint/eslint/issues/14198
             message: JSON.stringify(message),
             loc:
-              /* istanbul ignore next */ 'start' in place
+              /* istanbul ignore next */ place && 'start' in place
                 ? {
                     ...point,
                     start: { ...place.start, column: place.start.column - 1 },

